app/src/main/java/antidose/antidose/NotifyActivity.java

```diff
@@ -231,11 +231,19 @@ public void onResponse(Call<RestInterface.IncidentLocation> call, Response<RestI
                             float IncidentLatitude = response.body().getLatitude();
                             float IncidentLongitude = response.body().getLongitude();
 
-                            Intent intent = new Intent(NotifyActivity.this, NavigationActivity.class);
-                            intent.putExtra("incident-latitude", IncidentLatitude);
-                            intent.putExtra("incident-longitude", IncidentLongitude);
+                            if(Math.signum(IncidentLatitude)==0 && Math.signum(IncidentLongitude)==0){
+                                //then it has been cancelled
+                                Intent intent = new Intent(NotifyActivity.this, MainActivity.class);
+                                intent.putExtra("CANCEL_FRAGMENT", "TRUE");
+                                startActivity(intent);
 
-                            startActivity(intent);
+                            }else {
```

Comment:
Someone should line up the spacing/indentation at some point, but not for this PR.